openverse-api/catalog/api/views/audio_views.py

```diff
@@ -237,3 +244,33 @@ class AudioStats(MediaStats):
                          ])
     def get(self, request, format=None):
         return self._get(request, 'audio')
+
+
+class AudioWaveform(APIView):
+    swagger_schema = None
+
+    def get(self, request, identifier, format=None):
+        try:
+            audio = Audio.objects.get(identifier=identifier)
+        except Audio.DoesNotExist:
+            return Response(status=404, data='Audio not found')
+
+        file_name = None
+        try:
+            file_name = download_audio(audio.url, audio.identifier)
```

Comment:
Is there any chance `download_audio` could raise while writing to the temp directory, thereby never returning a `file_name` but still causing some need for cleanup?

Reply:
Yes, this is possible. We could consider possibly wiping out the `/tmp` directory periodically as a comprehensive cleanup solution?